core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -655,15 +655,19 @@ private[spark] object Utils extends Logging {
       // created the directories already, and that they are secured so that only the
       // user has access to them.
       getYarnLocalDirs(conf).split(",")
+    } else if (conf.getenv("SPARK_LOCAL_DIRS") != null) {
```

Comment:
@dragos do you know if people are currently using multiple directories on different volumes?
IMO we should probably try to make that possible via Mesos especially with persistent volumes introduced already, and not relying on users having a set place set with SPARK_LOCAL_DIRS.